### ISSUE TITLE
Replacing usage of 'eval' in repository

### ISSUE THREAD

**Issue by borgesa** [victoresque/pytorch-template#23]
Hi,

While I was trying to figure out how to both enable users to use both "PyTorch" loss functions and custom loss functions (ref. pull request #22), I was made aware that **usage of eval** often is discouraged and can even be dangerous (if abused).

Therefore, I propose that we replace the three instances of it in this repository:
model.py:
```python
def get_model_instance(model_arch, model_params):
    try:
        model = eval(model_arch)
```
metric.py
```python
def get_metric_functions(metric_names):
    try:
        metric_fns = [eval(metric) for metric in metric_names]
```
trainer.py
```python
   def _valid_epoch(self, epoch):
        self.model.eval()
```

I am not certain about what to replace it with. Any of you know?

**Comment by SunQpark** [comment]
The third example `model.eval()` seems to have nothing to do with this issue. 
The `eval()` discouraged to use may refer to the first and second, which changes string into the variable handle. I don't understand how this can be dangerous but I felt this uncomfortable since the linter always complains about that. Maybe I can study this later. I will post some references here, if I find.
I think we can just replace the `eval()` with use of **dictionary** of function handles.

**Comment by SunQpark** [comment]
From now on, it seems that simple usage of `eval()` just like ours has no problem.
Here is [reference](https://nedbatchelder.com/blog/201206/eval_really_is_dangerous.html)
